src/components/tryout/shorten-link.tsx

```diff
@@ -110,7 +112,24 @@ export function TryOutTab() {
             </div>
           </CardContent>
           <CardFooter>
-            <Button onClick={handleLinkShortenGeneration}>Shorten Link</Button>
+            <Button
+              onClick={handleLinkShortenGeneration}
+              className={`text-sm text-white bg-black hover:text-green-600 hover:bg-green-50 active:scale-95 active:ring-4 active:ring-green-300 transition-transform duration-300 ease-in-out ${
+                loading
+                  ? "bg-green-300 cursor-not-allowed hover:bg-green-300"
+                  : ""
+              }`}
+              // disabled={loading}
+            >
+              {loading ? (
+                <div className="flex">
+                  <span className="text-md text-black">loading...</span>
+                  <div className="ml-2 text-green-600 inline-block h-5 w-5 animate-spin rounded-full border-2 border-solid border-current border-r-transparent align-[-0.125em] motion-reduce:animate-[spin_1.5s_linear_infinite]"></div>
+                </div>
```

Comment:
Use the `Loader2` icon from Lucide-react icons so we avoid writing soo much code for a spinner

Reply:
Reference from shadcn website
<img width="771" alt="image" src="https://github.com/AmoabaKelvin/ishortn.ink/assets/97001695/3a9cf795-0105-4d7b-82a5-3f785886f788">

Reply:
> Use the `Loader2` icon from Lucide-react icons so we avoid writing soo much code for a spinner

Okay

Reply:
> Use the `Loader2` icon from Lucide-react icons so we avoid writing soo much code for a spinner

Okay